### PR TITLE
assert targets of type function are compared with R.identical

### PR DESCRIPTION
**Pull request by syves**
Added two tests for asserting that functions are compared using R.identical. Test names may need editing? Perhaps the adjustments to docs might need tweaking.

### CODE REVIEW

File: src/indexOf.js [hunk 1]
```diff
@@ -6,7 +6,8 @@ var _isArray = require('./internal/_isArray');
 /**
  * Returns the position of the first occurrence of an item in an array,
  * or -1 if the item is not included in the array. `R.equals` is used to
- * determine equality.
+ * determine equality in most cases. When target is a function `R.identical`
+ * is used to determine if the target is equal to an element in the array.
```
Comment (davidchambers) [inline src/indexOf.js]: This isn't quite right. `R.equals` is used to determine equality in all cases, but `R.equals` itself compares functions by identity. For example:

```javascript
var f = function() {};

R.equals(f, f);  // => true
R.equals(function() {}, function() {});  // => false
```

Perhaps we should clarify the description of `R.equals` instead.

File: test/indexOf.js [hunk 2]
```diff
@@ -83,4 +83,19 @@ describe('indexOf', function() {
...
+    var f = function() {};
+    var g = function() {};
+    var list = [g, f]
+    eq(R.indexOf(f, list), 1);
+  });
+
+    it('does not find function, compared by identity', function() {
+    var f = function() {};
+    var g = function() {};
+    var e = function() {};
```
Comment (davidchambers) [inline test/indexOf.js]: It's conventional to name the third function `h`. ;)

File: test/indexOf.js [hunk 3]
```diff
@@ -83,4 +83,19 @@ describe('indexOf', function() {
...
   });
+
+  it('finds function, compared by identity', function() {
+    var f = function() {};
+    var g = function() {};
+    var list = [g, f]
+    eq(R.indexOf(f, list), 1);
+  });
+
+    it('does not find function, compared by identity', function() {
```
Comment (davidchambers) [inline test/indexOf.js]: :arrow_left: :arrow_left:

**Comment by davidchambers** [comment]
The tests for `R.indexOf` look good. Could you add similar tests for `R.lastIndexOf`?

File: test/indexOf.js (hunk 2 above)
Comment (syves) [inline test/indexOf.js]: you are so conventional ;)

File: test/indexOf.js (hunk 3 above)
Comment (syves) [inline test/indexOf.js]: what does arrow arrow mean?

**Comment by syves** [comment]
added tests for lastIndexOf

File: test/indexOf.js (hunk 3 above)
Comment (scott-christopher) [inline test/indexOf.js]: I believe @davidchambers was suggesting the indentation is not aligned.

Comment (davidchambers) [inline test/indexOf.js]: Thanks for translating from Emoji to English, @scott-christopher. :)

File: src/indexOf.js [hunk 4]
```diff
@@ -21,6 +21,12 @@ var _isArray = require('./internal/_isArray');
  *
  *      R.indexOf(3, [1,2,3,4]); //=> 2
  *      R.indexOf(10, [1,2,3,4]); //=> -1
+ *
+ *      var f = function() {};
+ *      var g = function() {};
+ *      var h = fucntion() {};
```
Comment (davidchambers) [inline src/indexOf.js]: `s/fucntion/function/`

I wonder, though, whether we should include these examples on the website. They're very much worth including in the test suite, but the examples on the website should demonstrate common usage. Finding the index of a function in a list of functions is not a common operation, I would say. What do others think?

File: test/indexOf.js [hunk 5]
```diff
@@ -83,4 +83,19 @@ describe('indexOf', function() {
...
+    var f = function() {};
+    var g = function() {};
+    var h = function() {};
+    var list = [g, f];
+    eq(R.indexOf(h, list), -1);
+  });
 });
```
Comment (davidchambers) [inline test/indexOf.js]: Could you include an empty line between lines 100 and 101? :)

File: src/indexOf.js (hunk 4 above)
Comment (syves) [inline src/indexOf.js]: I'll default to you on this.

**Comment by syves** [comment]
@davidchambers can I have a tree?

File: src/indexOf.js [hunk 6]
```diff
@@ -21,6 +21,7 @@ var _isArray = require('./internal/_isArray');
  *
  *      R.indexOf(3, [1,2,3,4]); //=> 2
  *      R.indexOf(10, [1,2,3,4]); //=> -1
+ *
```
Comment (davidchambers) [inline src/indexOf.js]: :space_invader:

File: test/indexOf.js [hunk 7]
```diff
@@ -83,4 +83,20 @@ describe('indexOf', function() {
     var curried = R.indexOf(3);
     eq(curried(list), 2);
   });
+
+  it('finds function, compared by identity', function() {
+    var f = function() {};
+    var g = function() {};
+    var list = [g, f];
```
Comment (davidchambers) [inline test/indexOf.js]: You could make this `[g, f, g, f]` (as you do in the equivalent test for `R.lastIndexOf`) to assert that the first occurrence is matched.

**Comment by davidchambers** [comment]
> can I have a tree?

I left two comments about minor things. Address these and a `:deciduous_tree:` will be forthcoming. :)

File: src/indexOf.js (hunk 6 above)
Comment (syves) [inline src/indexOf.js]: @davidchambers does octopus mean remove space? Is there a guide to ramda emoji?

Comment (davidchambers) [inline src/indexOf.js]: There's no guide, I'm sorry! Hover over the emoji, though, and you will see its name – `:space_invader:`, in this case. ;)

**Comment by davidchambers** [comment]
:deciduous_tree:

**Comment by syves** [comment]
yay!

**Pull request merged**